pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenAssociatorInstancePaths`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the source instance.
+            If this object does specifies a namespace, it must equal the
+            value of the namespace parameter.
+            Its `host` attribute will be ignored.
+
+        See :meth:`~pywbem.AssociatorNames` for description of ResultClass,
+        Role, AssocClass, ResultClass, , FilterQueryLanguage, FilterQuery,
+        OperationTimeout, ContinueOnError, and MaxObjectCount parameters
+
+        Returns:
+
+            See :meth:`~pywbem..WBEMConnection.AssociatorNames` for description
+            of the returns
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+        self._test_pull_operations_disabled()
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout, ContinueOnError)
+
+        instances = self.AssociatorNames(namespace, InstanceName,
+                                         AssocClass=AssocClass,
+                                         ResultClass=ResultClass, Role=Role,
+                                         ResultRole=ResultRole)
+
+        return self._open_response(namespace, instances,
+                                   'PullInstancePaths',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
+
+    ####################################################################
+    #
+    #   OpenAssociatorInstances server request
+    #
+    ####################################################################
+
+    def OpenAssociatorInstances(self, namespace, InstanceName, AssocClass=None,
+                                ResultClass=None, Role=None, ResultRole=None,
+                                IncludeClassOrigin=None,
+                                PropertyList=None, FilterQueryLanguage=None,
+                                FilterQuery=None, OperationTimeout=None,
+                                ContinueOnError=None, MaxObjectCount=None):
+        """
+        Implements WBEM server responder for
+        WBEMConnection.OpenAssociatorInstances
+        with data from the instance store.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE